quinn-proto/src/endpoint.rs

```diff
@@ -341,7 +345,7 @@ where
         if self.is_full() {
             return Err(ConnectError::TooManyConnections);
         }
-        let remote_id = ConnectionId::random(&mut self.rng, MAX_CID_SIZE);
+        let (remote_id, _) = RandomConnectionIdGenerator::new(MAX_CID_SIZE).generate_cid();
```

Comment:
@liwenjieQu Is there any reason we hardcode the generator here, and don't use the `local_cid_generator`? The main difference I see between `RandomConnectionIdGenerator` vs `local_cid_generator` is the `cid_len`.

Here we pass `remote_id` to `add_connection`. However inside that function, we pass `self._local_cid_generator` to `Connection::new` which uses its length, instead of the length of `remote_id` (which was generated by `RandomConnectionIdGenerator`).

Should we change this to be `self.local_cid_generator.generate_cid()` for consistency.?

Reply:
Note the name `remote_id`; we're generating a *destination* connection ID here, not a local one. The client uses a random destination connection ID for its initial flight of packets; see https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#name-negotiating-connection-ids for details.

Reply:
@Ralith Thanks for the explanation! That clears it up.